docs/querying/datasource.md

```diff
@@ -370,3 +370,44 @@ always be correct.
 - Join algorithms other than broadcast hash-joins.
 - Join condition on a column compared to a constant value.
 - Join conditions on a column containing a multi-value dimension.
+
+### `unnest`
+
+Use the `unnest` datasource to unnest a column with multiple values in arrays. 
+For example, there is a source column that looks like this:
+
+| Nested | 
+| -- | 
+| [a, b] |
+| [c, d] |
+| [e, [f,g] |
+
+When you use the `unnest` datasource, the unnested column looks like this:
+
+| Unnested | 
+| -- |
+| a |
+| b |
+| c |
+| d |
+| e |
+| [f, g] |
+
+When unnesting data, keep the following in mind:
+
+- The total number of rows will grow to accommodate the new rows that the unnested data occupy. 
+- A single `unnest` datasource unnests one level of data. If you have multiple levels of nesting within a column, the `unnest` datasource, you need to combine multiple `unnest` datasources to fully unnest data that is nested in multiple layers. In the example, notice how the final column in **Nested** became two rows, `e` and `[f,g]` in **Unnested**.
+
+The `unnest` datasource uses the following syntax:
+
+```json
+"datasource": {
+  "type": "unnest",
+  "base": {
+    "type": "table",
+    "name": "example_table"
+  },
+  "column": "source_nested_column",
+  "outputName": "target_unnested_column"
```

Comment:
There has been one addition for `allowList` to filter over what we want to unnest